src/Params/ADnoteParameters.cpp

```diff
@@ -78,22 +78,16 @@ static const Ports voicePorts = {
     {"OscilSmp/", rDoc("Primary Oscillator"),
         &OscilGen::ports,
         rBOIL_BEGIN
-            if(obj->OscilGn == NULL) return;
+        if(obj->OscilGn == NULL) return;
         data.obj = obj->OscilGn;
-        SNIP
-            OscilGen::ports.dispatch(msg, data);
-        if(data.matches == 0)
-            data.forward();
+        data.forward();
```

Comment:
Now I'm seeing what you were mentioning about possible junk messages getting forwarded along. This approach is a-ok as long as there's the fix in middleware that was previously discussed in IRC.

Reply:
Yes, I just left most of the stuff like before. If you improvements for this callback, let me know.